addons/l10n_it/models/declaration_of_intent.py

```diff
@@ -0,0 +1,251 @@
+         ('revoked', 'Revoked'),
+         ('closed', 'Closed'),
+        ],
+        string="State",
+        readonly=True,
+        tracking=True,
+        default='draft',
+        help="The state of this Declaration of Intent. \n"
+        "- 'draft' means that the Declaration of Intent still needs to be confirmed before being usable. \n"
+        "- 'active' means that the Declaration of Intent is usable. \n"
+        "- 'closed' designates that the Declaration of Intent has been marked as not to use anymore without invalidating usages of it."
+        "- 'revoked' means the Declaration of Intent should not have been used. You will probably need to revert previous usages of it, if any.\n")
+
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        string='Company',
+        index=True,
+        required=True,
+        default=lambda self: self.env.company._accessible_branches()[:1],
+    )
+
+    partner_id = fields.Many2one(
+        comodel_name='res.partner',
+        string='Partner',
+        index=True,
+        required=True,
+        domain=lambda self: ['|', ('is_company', '=', True), ('parent_id', '=', False)],
+    )
+
+    currency_id = fields.Many2one(
+        comodel_name='res.currency',
+        string='Currency',
+        default=lambda self: self.env.ref('base.EUR').id,
+        required=True,
+    )
+
+    issue_date = fields.Date(
+        string='Date of Issue',
+        required=True,
+        copy=False,
+        default=fields.Date.context_today,
+        help="Date on which the Declaration of Intent was issued",
+    )
+
+    start_date = fields.Date(
+        string='Start Date',
+        required=True,
+        copy=False,
+        help="First date on which the Declaration of Intent is valid",
+    )
+
+    end_date = fields.Date(
+        string='End Date',
+        required=True,
+        copy=False,
+        help="Last date on which the Declaration of Intent is valid",
+    )
+
+    threshold = fields.Monetary(
+        string='Threshold',
+        required=True,
+        help="Total amount of services / goods you are allowed to sell without VAT under this Declaration of Intent",
+    )
+
+    consumed_amount = fields.Monetary(
+        string='Consumed Amount',
+        compute='_compute_consumed_amount',
+        store=True, readonly=True,
+        help="Total amount of sales of services / goods under this Declaration of Intent",
+    )
+
+    committed_amount = fields.Monetary(
+        string='Committed Amount',
+        compute='_compute_committed_amount',
+        store=True, readonly=True,
+        help="Total amount of planned sales of services / goods under this Declaration of Intent (i.e. quotations and sales orders)",
+    )
+
+    remaining_amount = fields.Monetary(
+        string='Remaining Amount',
+        compute='_compute_remaining_amount',
+        store=True, readonly=True,
+        help="Remaining amount (consumed or committed) to reach threshold",
```

Comment:
Remaining amount after deduction of the consumed and committed amounts.

Reply:
Done. I pushed a new version